ibis/expr/table_method.py

```diff
@@ -0,0 +1,75 @@
+import functools
```

Comment:
Could you please move the content of this file to `ibis.util`?

Reply:
Since this is tightly coupled with the expression system (for generating errors), I figured here was a better fit. It's not a general utility.